File: AssetManager/DataFunctions/LocalFunctions/DataConsistency.vb
Imports System.Text.RegularExpressions

Module DataConsistency
    Public ReadOnly strDBDateTimeFormat As String = "yyyy-MM-dd HH:mm:ss"
    Public ReadOnly strDBDateFormat As String = "yyyy-MM-dd"
    Public ReadOnly dtDefaultDate As Date = DateTime.Parse("1/1/0001 12:00:00 AM")

    Public Function NoNull(DBVal As Object) As String
        Try
            Return IIf(IsDBNull(DBVal), "", DBVal.ToString).ToString
        Catch ex As Exception
            ErrHandle(ex, System.Reflection.MethodInfo.GetCurrentMethod())
            Return ""
        End Try
    End Function

    ''' <summary>
    ''' Trims, removes LF and CR chars and returns a DBNull if string is empty.
    ''' </summary>
    ''' <param name="Value"></param>
    ''' <returns></returns>
    Public Function CleanDBValue(Value As String) As Object
        Dim CleanString As String = Regex.Replace(Trim(Value), "[/\r?\n|\r]+", String.Empty)
        Return IIf(CleanString = String.Empty, DBNull.Value, CleanString)
    End Function

    Public Function ValidPhoneNumber(PhoneNum As String) As Boolean
        If Trim(PhoneNum) <> "" Then
            Const nDigits As Integer = 10
            Dim fPhoneNum As String = ""
            Dim NumArray() As Char = PhoneNum.ToCharArray()
            For Each num As Char In NumArray
                If Char.IsDigit(num) Then fPhoneNum += num.ToString
            Next
            If Len(fPhoneNum) <> nDigits Then
                Return False
            Else
                Return True
            End If
        Else
            Return True
        End If
    End Function

    Public Function YearFromDate(dtDate As Date) As String
        Return dtDate.Year.ToString
    End Function

End Module

File: AssetManager.ConvertedToC#/Data/DataFunctions/ClientSideFunctions/DataConsistency.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Threading.Tasks;
using System.Text.RegularExpressions;
namespace AssetManager
{

	static class DataConsistency
	{

		public const string strDBDateTimeFormat = "yyyy-MM-dd HH:mm:ss";
		public static string NoNull(object DBVal)
		{
			try {
				return (Information.IsDBNull(DBVal) ? "" : DBVal.ToString()).ToString();
			} catch (Exception ex) {
				ErrorHandling.ErrHandle(ex, System.Reflection.MethodInfo.GetCurrentMethod());
				return "";
			}
		}

		/// <summary>
		/// Trims, removes LF and CR chars and returns a DBNull if string is empty.
		/// </summary>
		/// <param name="Value"></param>
		/// <returns></returns>
		public static object CleanDBValue(string Value)
		{
			string CleanString = Regex.Replace(Strings.Trim(Value), "[/\\r?\\n|\\r]+", string.Empty);
            if (CleanString == string.Empty)
            {
                return DBNull.Value;
            }
            else
            {
                return CleanString;
            }

          //  return (CleanString == string.Empty ? DBNull.Value : CleanString);
		}

		public static bool ValidPhoneNumber(string PhoneNum)
		{
			if (!string.IsNullOrEmpty(Strings.Trim(PhoneNum))) {
				const int nDigits = 10;
				string fPhoneNum = "";
				char[] NumArray = PhoneNum.ToCharArray();
				foreach (char num in NumArray) {
					if (char.IsDigit(num))
						fPhoneNum += num.ToString();
				}
				if (Strings.Len(fPhoneNum) != nDigits) {
					return false;
				} else {
					return true;
				}
			} else {
				return true;
			}
		}

		public static string YearFromDate(System.DateTime dtDate)
		{
			return dtDate.Year.ToString();
		}

		public static string DeviceHostnameFormat(string Serial)
		{
			return "D" + Strings.Trim(Serial);
		}

		public static bool IsValidYear(string Year)
		{
			try {
				if (!string.IsNullOrEmpty(Strings.Trim(Year))) {
					if (Enumerable.Range(1900, 200).Contains(Convert.ToInt32(Year))) {
						return true;
					}
				}
				return false;
			} catch {
				return false;
			}
		}

	}
}
